src/tests/Microsoft.Diagnostics.Monitoring.EventPipe/EventCounterPipelineUnitTests.cs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.NETCore.Client.UnitTests;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Extensions;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe.UnitTests
+{
+    public class EventCounterPipelineUnitTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public EventCounterPipelineUnitTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        private sealed class TestMetricsLogger : IMetricsLogger
+        {
+            private readonly ITestOutputHelper _output;
+            private Dictionary<string, ICounterPayload> _metrics = new Dictionary<string, ICounterPayload>();
+
+            public TestMetricsLogger(ITestOutputHelper output)
+            {
+                _output = output;
+            }
+
+            public void Dispose()
+            {
+            }
+
+            public IEnumerable<ICounterPayload> Metrics => _metrics.Values;
+
+            public void LogMetrics(ICounterPayload metric)
+            {
+                _metrics[string.Concat(metric.GetProvider(), "_", metric.GetName())] = metric;
+            }
+
+            public void PipelineStarted()
+            {
+            }
+
+            public void PipelineStopped()
+            {
+            }
+        }
+
+        [SkippableFact]
+        public async Task TestCounterEventPipeline()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                throw new SkipTestException("Unstable test on OSX");
```

Comment:
@hoyosjs and I spent a lot of time trying to figure out the issue on OSX. The error is very obscure and we never got to the bottom of it. I can try this again.

Reply:
This has been a testing issue for a long time, setting up a separate test process and egressing data. This is a test instability rather than a new regression in existing code.

Reply:
That sounds modestly less scary, though it still sounds like we wouldn't be getting any test coverage on OSX? Given that we don't disable any of our other diagnostics tooling tests on OSX (that I am aware of? maybe we have preexisting gaps?) it still seems suspicious why these tests would be problematic and none of the others are.

Reply:
When I first did some of this work I attempted to add a more functional style test which uses EventSourceLogger in a debugee process and coordinates with the test process. This worked great on Linux and Windows, and had obscure errors in OsX. I understand that you are worried about regressions, but as far as I know it's a test issue.

Reply:
I reenabled tests for OSX and so far they seem stable.